FullStorySegmentMiddleware/Classes/FullStoryMiddleware.m

```diff
@@ -64,11 +64,11 @@ - (void)context:(SEGContext * _Nonnull)context next:(SEGMiddlewareNext _Nonnull)
                 SEGTrackPayload *payload = (SEGTrackPayload *)ctx.payload;
 
                 // transform props to comply with FS custom events requirement
-                NSDictionary *props = [self getSuffixedProps:payload.properties];
-                NSLog(@"%@",props);
+                FSSuffixedProperties *fsProps = [[FSSuffixedProperties alloc] initWithProperties:payload.properties];
+
                 // Segment Track event, optionally enabled /w events allowlisted, send as custom events into FullStory
                 if (self.allowlistAllTrackEvents || [self.allowlistEvents containsObject:payload.event]) {
-                    [FS event:payload.event properties:props];
+                    [FS event:payload.event properties:fsProps.suffixedProperties];
```

Comment:
Since `fsProps` is only used inside the if statment, you should probably move the line:
```
FSSuffixedProperties *fsProps = [[FSSuffixedProperties alloc] initWithProperties:payload.properties];
```
inside the if statement as well. That way you're not computing the suffixed properties for the cases where you don't take the if.